src/lib/operators/projection.cpp

```diff
@@ -97,47 +97,109 @@ std::shared_ptr<const Table> Projection::_on_execute() {
     step_performance_data.set_step_runtime(OperatorSteps::UncorrelatedSubqueries, timer.lap());
   }
 
+  auto forwarding_cost = std::chrono::nanoseconds{};
+  auto expression_evaluator_cost = std::chrono::nanoseconds{};
+
+  const auto chunk_count = input_table.chunk_count();
+
   // Perform the actual projection on a per-chunk level. `output_segments_by_chunk` will contain both forwarded and
   // newly generated columns. In the upcoming loop, we do not yet deal with the projection_result_table indirection
   // described above.
-  auto output_segments_by_chunk = std::vector<Segments>(input_table.chunk_count());
+  auto output_segments_by_chunk = std::vector<Segments>(chunk_count);
 
-  auto forwarding_cost = std::chrono::nanoseconds{};
-  auto expression_evaluator_cost = std::chrono::nanoseconds{};
+  // Create a vector that will hold all jobs that are getting executed in parallel.
+  auto jobs = std::vector<std::shared_ptr<AbstractTask>>{};
+  jobs.reserve(chunk_count);
```

Comment:
I like the amount of documentation in this PR, so I'm somewhat hesitant to say this: This particular comment is somewhat redundant.